Modules/mathmodule.c

```diff
@@ -2068,11 +2068,6 @@ math_trunc(PyObject *module, PyObject *x)
         return PyFloat_Type.tp_as_number->nb_int(x);
     }
 
-    if (!_PyType_IsReady(Py_TYPE(x))) {
```

Comment:
Why is it removed? Are you sure that it is no longer needed for all third-party types?

Reply:
If so, is there reason why there is no such checks for ``math_ceil`` or ``math_trunc``?

Reply:
I do not know, perhaps it should be added there. Look at the history of the code, read previous related discussions. Open a new issue for this change, it may be not so innocent.

See https://en.wikipedia.org/wiki/G._K._Chesterton#Chesterton's_fence .

Reply:
> Look at the history of the code

I did that (as good as I can).  This seems to be a history artifact: this function was introduced in 400adb030a as a moved ``__builtins__.trunc()``, while ``math.ceil/floor`` were here from the initial revision...

> Open a new issue for this change, it may be not so innocent.

Lets wait for a while if @mdickinson could take a look on that part.

Reply:
I think this can be marked as resolved, see #111417